src/main.js

```diff
@@ -284,22 +284,26 @@ function calculateStats(modules) {
                 _Student.gradesCount[parsedModule.grade]++;
                 _Student.totalGrades++;
             }
+
+            let credits;
             if (parsedModule.passed) {
-                let credits = Number(parsedModule.credits);
+                credits = Number(parsedModule.credits);
```

Comment:
What is the intended behaviour here? If credits is NaN, then the total credits Calculation breaks completely, as 
```_Student.totalCredits += credits // credits = NaN```
would result in  `_Student.totalCredits` being NaN. Also `NaN > 0 // is false` Therefore it would not enter the if clause below. I am unsure about the intended behaviour. I could change line 
```credits = Number(parsedModule.credits);```
to
```credits = Number(parsedModule.credits) || 0```

Would that suffice?

Reply:
No, the problem is that you only count the credits, if you have passed the module. If you have a module with and F or a grade below 4, the variable `credits` is never parsed and isn't included into the calculation. The easiest would be to move `credits = Number(parsedModule.credits);` outside the if clause. Correct me if I made a mistake.